site/pages/admin/user-management.page.tsx

```diff
@@ -49,21 +51,51 @@ const UserManagement = ({ userList }: UserManagementPageProps): ReactElement =>
         return rows;
     };
 
-    const createLink = (key: string, index?: number, sendInvite?: boolean) => {
-        return sendInvite ? (
+    const removeUser = () => {
+        setShowRemovePopup(!showRemovePopup);
+    };
+
+    const cancelActionHandler = () => {
+        setShowRemovePopup(false);
+    };
+
+    const createLink = (key: string, index: number, sendInvite?: boolean) => {
+        return (
             <>
-                <Link key={`${key}${index ? `-${index}` : ""}`} className="govuk-link" href="/">
-                    Resend invite
-                </Link>
-                <br />
-                <Link key={`${key}${index ? `-remove-${index}` : "-remove"}`} className="govuk-link" href="/">
-                    Remove
-                </Link>
+                {showRemovePopup ? (
+                    <DeleteConfirmationPopup
```

Comment:
This is creating a modal for each link so it will actually be rendering multiple modals on top of each other and the top one will always be for the last user in the list, the modal should be moved out to the top level and the username to be deleted will need to be set in state, something like:

```ts
const UserManagement = ({ userList, csrfToken }: UserManagementPageProps): ReactElement => {
    const numberOfUserPages = Math.ceil(userList.length / 10);
    const [currentPage, setCurrentPage] = useState(1);
    const [userToDelete, setUserToDelete] = useState<string | null>(null);

    const getAccountType = (groupName: UserGroups): string => {
        switch (groupName) {
            case UserGroups.systemAdmins:
            case UserGroups.orgAdmins:
                return "Admin";
            case UserGroups.orgPublishers:
                return "Publisher";
            case UserGroups.orgStaff:
                return "Staff";
        }
    };

    const getRows = () => {
        const rows: { header?: string | ReactNode; cells: string[] | ReactNode[] }[] = [];
        getDataInPages(currentPage, userList).forEach((user, index) => {
            rows.push({
                cells: [
                    `${getAccountType(user.group)}`,
                    user.email,
                    user.userStatus === "CONFIRMED" ? "Active" : "Pending invite",
                    createLink("user-action", index, user.username, user.userStatus === "CONFIRMED" ? false : true),
                ],
            });
        });
        return rows;
    };

    const removeUser = (username: string) => {
        setUserToDelete(username);
    };

    const cancelActionHandler = () => {
        setUserToDelete(null);
    };

    const createLink = (key: string, index: number, username: string, sendInvite?: boolean) => {
        return (
            <>
                {sendInvite ? (
                    <>
                        <Link key={`${key}${index ? `-${index}` : ""}`} className="govuk-link" href="/">
                            Resend invite
                        </Link>
                        <br />
                        <button
                            key={`${key}${index ? `-remove-${index}` : "-remove"}`}
                            className="govuk-link"
                            onClick={() => removeUser(username)}
                        >
                            Remove
                        </button>
                    </>
                ) : (
                    <button
                        key={`${key}${index ? `-${index}` : ""}`}
                        className="govuk-link"
                        onClick={() => removeUser(username)}
                    >
                        Remove
                    </button>
                )}
            </>
        );
    };

    return (
        <BaseLayout title={title} description={description}>
            <>
                {userToDelete ? (
                    <DeleteConfirmationPopup
                        entityName="user"
                        deleteUrl="/api/admin/delete-user"
                        cancelActionHandler={cancelActionHandler}
                        csrfToken={csrfToken || ""}
                        hiddenInputs={[
                            {
                                name: "username",
                                value: userToDelete,
                            },
                        ]}
                    />
                ) : null}
                <div className="govuk-form-group">
                    <h1 className="govuk-heading-xl">User Management</h1>
                </div>
                <Table
                    caption={{ text: "Users", size: "m" }}
                    columns={["Account type", "User email", "Status", "Action"]}
                    rows={getRows()}
                ></Table>
                <Link role="button" href={"/admin/add-user"} className="govuk-button--secondary govuk-button mt-5">
                    Add new user
                </Link>
                <PageNumbers
                    numberOfPages={numberOfUserPages}
                    currentPage={currentPage}
                    setCurrentPage={setCurrentPage}
                />
            </>
        </BaseLayout>
    );
};
```

Reply:
done